--- ebuild.sh.orig	Thu Dec  6 22:24:34 2001
+++ ebuild.sh	Thu Dec  6 23:04:27 2001
@@ -1,4 +1,8 @@
 #!/bin/bash
+
+# Disable the SandBox for now ...
+export SANDBOX_ON="0"
+
 #we need this next line for "die" and "assert"
 shopt -s expand_aliases
 source /etc/profile.env > /dev/null 2>&1
@@ -694,7 +698,25 @@
 		  set +x
 		fi
 	    ;;
-	unpack|compile|help|touch|clean|fetch|digest|pkginfo|pkgloc|unmerge|merge|package|install|rpm)
+	# Only enable the SandBox for these functions
+	unpack|compile|clean|install)
+		if [ ${SANDBOX_DISABLED="0"} = "0" ]
+		then
+			export SANDBOX_ON="1"
+		else
+			export SANDBOX_ON="0"
+		fi
+		if [ "$PORTAGE_DEBUG" = "0" ]
+		then
+			dyn_${myarg}
+		else
+			set -x
+			dyn_${myarg}
+			set +x
+		fi
+		export SANDBOX_ON="0"
+		;;
+	help|touch|fetch|digest|pkginfo|pkgloc|unmerge|merge|package|rpm)
 	    if [ "$PORTAGE_DEBUG" = "0" ]
 	    then
 	      dyn_${myarg}
