--- portage.py.orig	Thu Dec  6 22:24:34 2001
+++ portage.py	Thu Dec  6 23:10:11 2001
@@ -525,11 +525,14 @@
 	global settings
 	mypid=os.fork()
 	if mypid==0:
-		mycommand="/bin/bash"
+#		mycommand="/bin/bash"
+		mycommand="/usr/bin/sandbox"
 		if debug:
-			myargs["bash","-x","-c",mystring]
+#			myargs["bash","-x","-c",mystring]
+			myargs=["sandbox",mystring]
 		else:
-			myargs=["bash","-c",mystring]
+#			myargs["bash","-c",mystring]
+			myargs=["sandbox",mystring]
 		os.execve(mycommand,myargs,settings.environ())
 		return
 	retval=os.waitpid(mypid,0)[1]
@@ -559,6 +562,7 @@
 	#PEBUILD
 	settings["FILESDIR"]=settings["O"]+"/files"
 	settings["PF"]=os.path.basename(settings["EBUILD"])[:-7]
+	settings["SANDBOX_LOG"]=settings["PF"]
 	mysplit=pkgsplit(settings["PF"],0)
 	if mysplit==None:
 		print "!!! Error: PF is null; exiting."
